yarnitor/background/worker.py

```diff
@@ -296,21 +430,43 @@ def _make_application_handler(self, yarn_application_info):
         """
         app_type = yarn_application_info['applicationType']
         klass = self.application_handlers.get(app_type, BaseHandler)
-        return klass.from_yarn_application_info(yarn_application_info)
+        return klass(yarn_application_info['trackingUrl'], yarn_application_info['id'])
 
     def _generate_listing(self):
-        """Computes the listing of applications and the additional information
+        """Computes the listing of YARN applications and the additional information
         provided by the handlers.
 
+        Blocks while running the work of fetching application details from tracking
+        APIs using the global threadpool.
+
+        Returns
+        -------
+        dict
+            YARN application IDs as keys mapped to application detail dictionaries
+            as values
         """
+        # Fetch all running applications
         cluster_apps = self.yarn_handler.cluster_applications("RUNNING")
         if 'apps' not in cluster_apps or cluster_apps['apps'] is None:
+            # Something might be wrong if there are no applications
             logger.warn('No application data available')
```

Comment:
should this be an error? My gut reaction is that if there's no app data something is broken somewhere

Reply:
Maybe. It's also possible it means there are simply no applications. It's not well documented in the YARN API whether the keys are present when there are no apps or not.